docs/core/preview3/tools/dotnet-build.md

```diff
@@ -23,33 +23,37 @@ dotnet-build -- Builds a project and all of its dependencies
 
 ## Synopsis
 
-`dotnet build [--help] [--output]  [--framework]  
-    [--configuration]  [--runtime] [--version-suffix]
-    [--build-profile]  [--no-incremental] [--no-dependencies]
-    [<project>]`
+```
+dotnet build <project>
+dotnet build [-h|--help]
+dotnet build [-o|--output] [-f|--framework] [-c|--configuration] [-r|--runtime]
+dotnet build [--version-suffix] [--no-incremental] [--no-dependencies]
+```
+
+
 
 ## Description
+The `dotnet build` command builds the project and its dependencies into a set of binaries. The binaries are the symbol files used for debugging (having a `*.pdb` extension) as well as the project's code in Intermediate Language (IL) with an `*.dll` extension. Additionally, a JSON file that lists out the dependencies of the application with the `*.deps` extension will be produced. Finally, a `runtime.config.json` file will be produced as well. This file specifies which shared runtime and version the built code will run against. 
+
+If the project has third-party dependencies, such as libraries from NuGet, these will be resolved from the NuGet cache and will not be available with the project's built output. With that in mind, the product of `dotnet build` is not ready to be transferred to another machine to run. This is in contrast to the behavior of .NET Framework in which building an executable project (an application) will produce an output that is possible to run on any machine that has .NET Framework installed. In order to get a similar experience in .NET Core, you have to use the [dotnet publish](dotnet-publish.md) command. More information about this can be found in the [.NET Core Application Deployment](../deploying/index.md) document. 
```

Comment:
Some suggested rewritings:

With that in mind, the product of `dotnet build` is not ready to be transferred to another machine to run. -> This means that the product of `dotnet build` should not be used for deploying the app to other machines.

This is in contrast to the behavior of .NET Framework in which building an executable project (an application) will produce an output that is possible to run on any machine that has .NET Framework installed. ->
This is an important distinction to make for people that are used with the .NET Framework behavior, where building an application will produce an output that can be run on any machine that has .NET Framework installed.

In order to get a similar experience in .NET Core, -> In order to get your app ready to be deployed in .NET Core,

More information about this can be found in the [.NET Core Application Deployment](../deploying/index.md) document. -> For more information, see the [.NET Core Application Deployment](../deploying/index.md) article. (it's kind of a boilerplate sentence)